ksmt-runner/src/main/kotlin/org/ksmt/solver/runner/KSolverRunnerExecutor.kt

```diff
@@ -24,66 +33,138 @@ import org.ksmt.solver.KSolverStatus
 import org.ksmt.solver.KSolverUnsupportedFeatureException
 import org.ksmt.solver.KSolverUnsupportedParameterException
 import org.ksmt.solver.model.KModelImpl
+import org.ksmt.solver.runner.KSolverRunnerManager.CustomSolverInfo
 import org.ksmt.sort.KBoolSort
 import org.ksmt.sort.KSort
 import org.ksmt.sort.KUninterpretedSort
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
 import kotlin.time.Duration
 
 class KSolverRunnerExecutor(
     private val hardTimeout: Duration,
     private val worker: KsmtWorkerSession<SolverProtocolModel>,
 ) {
+    fun configureSync(config: List<SolverConfigurationParam>) {
+        ensureActive()
+
+        queryWithTimeoutAndExceptionHandlingSync {
+            configure.querySync(config)
+        }
+    }
 
     suspend fun configureAsync(config: List<SolverConfigurationParam>) {
         ensureActive()
 
-        queryWithTimeoutAndExceptionHandling {
-            configure.startSuspending(worker.lifetime, config)
+        queryWithTimeoutAndExceptionHandlingAsync {
+            configure.queryAsync(config)
+        }
+    }
+
+    fun assertSync(expr: KExpr<KBoolSort>) {
+        ensureActive()
+
+        val params = AssertParams(expr)
+        queryWithTimeoutAndExceptionHandlingSync {
+            assert.querySync(params)
         }
     }
 
     suspend fun assertAsync(expr: KExpr<KBoolSort>) {
         ensureActive()
 
         val params = AssertParams(expr)
-        queryWithTimeoutAndExceptionHandling {
-            assert.startSuspending(worker.lifetime, params)
+        queryWithTimeoutAndExceptionHandlingAsync {
+            assert.queryAsync(params)
+        }
+    }
+
+    fun assertAndTrackSync(expr: KExpr<KBoolSort>, trackVar: KConstDecl<KBoolSort>) {
+        ensureActive()
+
+        val params = AssertAndTrackParams(expr, trackVar)
+        queryWithTimeoutAndExceptionHandlingSync {
+            assertAndTrack.querySync(params)
         }
     }
 
     suspend fun assertAndTrackAsync(expr: KExpr<KBoolSort>, trackVar: KConstDecl<KBoolSort>) {
         ensureActive()
 
         val params = AssertAndTrackParams(expr, trackVar)
```

Comment:
The same question about duplicated code. I don't mind to have this code, but you made some work to eliminate it in other file, is it possible to do it here?